Shift3/boilerplate-client-react
# Added UI views for resetPassword, changePassword, forgotPassword, logIn, & signUp

Part of #19 
## Changes

1. Updated UI for the following components, at the designated route:
- `logInForm` and `logInPage` at "/auth/login"
- `signUpForm` and `signUpPage` at "/auth/signup"
- `changePasswordForm` and `changePasswordPage` at "/auth/change-password"
- `resetPasswordForm` and `resetPasswordPage` at "/auth/reset-password" 
- `forgotPasswordForm` and `forgotPasswordPage` at "/auth/forgot-password"

2. Added unit tests to above components for logic, schema, and styled components with:
`index.spec.tsx` 
`styled.spec.tsx`
`schema.spec.tsx`

## Approach
To create UI to reflect the Angular Boilerplate `https://boilerplate-client-angular.shift3sandbox.com/auth/login`, while also adding unit tests for each component.

`React-Testing-Library` along with `React-Test-Renderer` were use for all unit tests. For more information visit: 
'https://testing-library.com/docs/react-testing-library/intro/'
'https://reactjs.org/docs/test-renderer.html'

## Testing
1. Pull in the changes to your local copy of this branch and run `yarn start`
2. Check that the app renders.
3. Run `yarn test` and check that all tests are passing.

## Screenshots
![Screen Shot 2021-06-14 at 10 17 18 PM](https://user-images.githubusercontent.com/59614789/121982863-5ca6fe80-cd5e-11eb-8a7a-7a9f0e493817.png)

### Closes #181 #199
